### PR TITLE
Fix loki_server version property

### PR DESCRIPTION
**Pull request by Abuelodelanada**
We were hardcoding Loki version because a Loki bug. This bug was solved.

### CODE REVIEW

File: src/loki_server.py [hunk 1]
```diff
@@ -65,16 +65,8 @@ def version(self) -> str:
...
-            # We are hardcoding version here because there is and bug in last stable Loki version
-            # (2.3.0) that do not return its version: https://github.com/grafana/loki/issues/4133
-            # This bug it was already addressed in PR: https://github.com/grafana/loki/pull/4135
-            # but it's note released yet.
-            #  info = self._build_info()
-            #  return = info.get("version", None)
-            _ = self._build_info()
-            return "2.3.0"
+            info = self._build_info()
+            return info.get("version", None)
```
Comment (sed-i) [inline src/loki_server.py:69]: Given the `None` is preferred to a KeyError, maybe move the return below the try-except block?

**Review by sed-i — APPROVED** [review]

File: src/loki_server.py (hunk 1 above)
Comment (Abuelodelanada) [reply src/loki_server.py:69]: Yes, absolutely

File: src/loki_server.py [hunk 2]
```diff
@@ -65,21 +65,17 @@ def version(self) -> str:
...
-            # This bug it was already addressed in PR: https://github.com/grafana/loki/pull/4135
-            # but it's note released yet.
-            #  info = self._build_info()
-            #  return = info.get("version", None)
-            _ = self._build_info()
-            return "2.3.0"
+            info = self._build_info()
+            version = info.get("version", None)
+            if not version:
+                raise LokiServerNotReadyError("Loki version could not be retrieved.")
```
Comment (balbirthomas) [inline src/loki_server.py:71]: Only reason I can think that the version will not be available is that the Loki server is not yet responding to queries. If this is true then I would expect that the Loki charm sets it status to waiting status or defer and event handler depending on the use case for which version is being retrieved. I am not too familiar with the whole charm so I am leaving this to your better judgement.

**Review by balbirthomas — APPROVED** [review]

File: src/loki_server.py (hunk 2 above)
Comment (Abuelodelanada) [reply src/loki_server.py:71]: Yes, that's why we raise LokiServerNotReadyError

**Pull request merged**